src/ore/src/vec.rs

```diff
@@ -29,7 +29,7 @@ pub fn repurpose_allocation<T1, T2>(mut v: Vec<T1>) -> Vec<T2> {
     let cap = v.capacity();
     let p = v.as_mut_ptr().cast();
     std::mem::forget(v);
-    // This is safe because `T1` and `T2` have the same size and alignment,
+    // This is sound because `T1` and `T2` have the same size and alignment,
```

Comment:
The stdlib also uses safe to describe this but I'm ok with this PR https://doc.rust-lang.org/src/std/sys/unix/os_str.rs.html#142